src/js/builtins/JSBufferPrototype.ts

```diff
@@ -353,49 +463,66 @@ export function writeUIntBE(this: BufferExt, value, offset, byteLength) {
       view.setUint16(offset, Math.floor(value * 2 ** -32), false);
       break;
     }
-    default: {
-      throw new RangeError("byteLength must be >= 1 and <= 6");
-    }
   }
   return offset + byteLength;
 }
 
 export function writeFloatLE(this: BufferExt, value, offset) {
+  if (offset === undefined) offset = 0;
+  require("internal/validators").validateInteger(offset, "offset", 0, this.byteLength - this.byteOffset - 4);
+  if (this.byteOffset + offset + 4 > this.byteLength) throw require("internal/errors").ERR_BUFFER_OUT_OF_BOUNDS();
```

Comment:
Ditto here in all of these re: validation

Also, when you see us access the same property more than once, it's better to assign it to a local variable.